packages/core/src/shared/clients/ec2.ts

```diff
@@ -31,14 +31,18 @@ import { ClientWrapper } from './clientWrapper'
 import { AsyncCollection } from '../utilities/asyncCollection'
 
 /**
- * A wrapper around EC2.Instance where we can safely assume InstanceId field exists.
+ * A wrapper around Instance where we can safely assume InstanceId field exists.
  */
-export interface SafeEc2Instance extends Instance {
+export interface PatchedEc2Instance extends Instance {
```

Comment:
would we have some other class for "unsafe"? or is that just an improvement that we are making in our wrapper class?

```suggestion
export interface Ec2Instance extends Instance {
```

Reply:
This is an improvement here to make processing the responses easier. Avoids undefined checks in multiple places since we assert it once, and store that info in the type.